src/com/amazon/ion/impl/bin/IonManagedBinaryWriter.java

```diff
@@ -740,16 +740,22 @@ public int getDepth()
 
     private void startLocalSymbolTableIfNeeded(final boolean writeIVM) throws IOException
     {
-        if (symbolState == SymbolState.SYSTEM_SYMBOLS)
+        boolean isAppend = symbolState == SymbolState.LOCAL_SYMBOLS_FLUSHED;
+        if (symbolState == SymbolState.SYSTEM_SYMBOLS || isAppend)
         {
-            if (writeIVM)
+            if (writeIVM && !isAppend)
             {
                 symbols.writeIonVersionMarker();
             }
             symbols.addTypeAnnotationSymbol(systemSymbol(ION_SYMBOL_TABLE_SID));
             symbols.stepIn(STRUCT);
             {
```

Comment:
maybe I'm missing something but this block wrapping seems unnecessary. You could remove it and save a bit on indentation

Reply:
It's not necessary, it's a style choice that's been made by some ion-java authors in the past to create a new indentation block when stepping into a writer so that the writer code mimics the depth of the encoding. If we want to clean that up in a future style-focused PR we can do that.